apps/champions/test/battle_test.exs

```diff
@@ -679,4 +679,69 @@ defmodule Champions.Test.BattleTest do
                Champions.Battle.Simulator.run_battle([unit], [target_dummy_5], maximum_steps: maximum_steps).result
     end
   end
+
+  test "All", %{target_dummy_character: target_dummy_character} do
+    maximum_steps = 1
+
+    # Create a character with a basic skill that will deal 10 damage to all the enemies
+    basic_skill_params =
+      TestUtils.build_skill(%{
+        name: "DealDamage All Enemies",
+        mechanics: [
+          %{
+            trigger_delay: 0,
+            apply_effects_to:
+              TestUtils.build_apply_effects_to_mechanic(%{
+                effects: [
+                  TestUtils.build_effect(%{
+                    executions: [
+                      %{
+                        type: "DealDamage",
+                        attack_ratio: 1,
+                        energy_recharge: 0
+                      }
+                    ]
+                  })
+                ],
+                targeting_strategy: %{
+                  type: "all",
+                  target_allies: false
+                }
+              })
+          }
+        ],
+        cooldown: maximum_steps * @miliseconds_per_step - 1
+      })
+
+    {:ok, character} =
+      TestUtils.build_character(%{
+        name: "Character",
+        basic_skill: basic_skill_params,
+        ultimate_skill: TestUtils.build_skill(%{name: "Empty Skill"}),
+        base_attack: 10,
+        base_health: 10
+      })
+      |> Characters.insert_character()
+
+    {:ok, unit} = TestUtils.build_unit(%{character_id: character.id}) |> Units.insert_unit()
+    {:ok, unit} = Units.get_unit(unit.id)
+
+    # Create 6 target dummies for enemy team
+    [target_dummy_1, target_dummy_2, target_dummy_3, target_dummy_4, target_dummy_5, target_dummy_6] =
```

Comment:
Let's just capture this whole list as `target_dummies`

Reply:
Fixed!